lib/src/client.dart

```diff
@@ -30,6 +30,13 @@ abstract class Client {
   /// [BrowserClient] if `dart:html` is available.
   factory Client() => new IOClient();
 
+  /// Creates a new client handler.
+  ///
+  /// The [handler] is a function that receives a [Request] and returns a
+  /// [Future<Response>]. It will be called when [Client.send] is invoked. This
+  /// allows for composition of a [Client] within a larger application.
```

Comment:
This last sentence is confusing. It's not clear to me why this constructor in particular makes composition easier.